Frontend/ui-customer/src/Screens/CustomerUtils/ReportIssueChatScreen/Controller.purs

```diff
@@ -0,0 +1,475 @@
+            | ChatViewActionController ChatView.Action
+            | SendMessage ChatView.ChatComponentConfig Boolean
+            | UpdateState ReportIssueChatScreenState
+            | PrimaryEditTextActionController PrimaryEditText.Action
+            | ReopenIssuePress
+            | KeyboardCallback String
+            | InitializeCallback
+
+data ScreenOutput = GoBack
+                  | UploadIssue  ReportIssueChatScreenState
+                  | CallDriver ReportIssueChatScreenState
+                  | CallSupport ReportIssueChatScreenState
+                  | SelectIssueOption ReportIssueChatScreenState
+                  | ReopenIssue ReportIssueChatScreenState
+
+eval :: Action -> ReportIssueChatScreenState -> Eval Action ScreenOutput ReportIssueChatScreenState
+
+eval ReopenIssuePress state = 
+  exit $ ReopenIssue state
+
+eval (Exit output) state =
+  exit output
+
+eval (KeyboardCallback keyBoardState) state = do 
+  let keyState = case keyBoardState of
+                    "onKeyboardOpen" -> true
+                    "onKeyboardClose" -> false
+                    _ -> false 
+  _ <- pure $ scrollToEnd (getNewIDWithTag "ChatScrollView") true
+  continue state {props {isKeyboardOpen = keyState}}
+
+eval BackPressed state =
+  continueWithCmd state [do
+    _ <- runEffectFn1 removeMediaPlayer ""
+    _ <- pure $ hideKeyboardOnNavigation true
+    pure $ Exit GoBack
+  ]
+
+eval AfterRender state =
+  continue state
+
+eval SubmitIssue state =
+  exit $ UploadIssue state
+
+eval ShowOptions state = do
+  let options'  = map (\x -> x.option) state.data.options
+  continue state { data { chatConfig { enableSuggestionClick = false, suggestionsList = options' } } }
+
+---------------------------------------------------- Add Media ----------------------------------------------------
+eval AddImage state =
+  if length state.data.addedImages > 0
+  then
+    continueWithCmd state { props { showImageModel = true, isPopupModelOpen = true }
+                   , data  { addImagesState { images = state.data.addedImages, stateChanged = false } } } [do
+                     void $ runEffectFn1 clearFocus (getNewIDWithTag "submit_chat_edit_text")
+                     pure NoAction
+                   ]
+  else
+    continueWithCmd state { props { showImageModel = true, isPopupModelOpen = true }
+                          , data  { addImagesState { images = state.data.addedImages, stateChanged = false } } } [do
+      void $ runEffectFn1 clearFocus (getNewIDWithTag "submit_chat_edit_text")
+      void $ pure $ startLottieProcess lottieAnimationConfig{ rawJson = "primary_button_loader.json", lottieId = (getNewIDWithTag "add_images_model_done_button"), scaleType = "CENTER_CROP" }
+      pure NoAction
+    ]
+
+eval (AddAudio stateChangedState) state = do
+  _ <- pure $ hideKeyboardOnNavigation true
+  if isJust state.data.recordedAudioUrl
+  then
+    continueWithCmd state { props { showAudioModel = true, isPopupModelOpen = true }
```

Comment:
Looks like some state is similar in both cases
Create a new var with common state and use it

Reply:
no. both are not similar.